internal/consul/leader.go

```diff
@@ -23,17 +23,20 @@ import (
 	"github.com/hashicorp/consul/api"
 )
 
-func AcquireLock(logger log.Logger, client *Client, consulSession *Session) error {
-	isLeader, _, err := client.ConsulClient.KV().Acquire(&api.KVPair{
-		Key:     consulSession.Name,
-		Value:   []byte(consulSession.ID),
-		Session: consulSession.ID,
+func (c *Client) AcquireLock(key string) error {
+	isLeader, _, err := c.underlying.KV().Acquire(&api.KVPair{
+		Key:     key,
+		Value:   []byte(c.session.ID),
+		Session: c.session.ID,
 	}, nil)
 	if err != nil {
 		return err
 	}
 	if isLeader {
 		return nil
 	}
-	return logger.Info().LogErrorf("%s is not the leader", client.NodeId).Err()
+	return c.logger.Error().With(log.Fields{
```

Comment:
Should this be an Error?  Won't hosts always try to check if they can be the leader and this should just inform it that there's already a leader?

Reply:
Oh yea, forgot to set that back as Info.

Reply:
Ignore my comment, I see that we're returning nil if the host can acquire a lock.  This makes sense to me actually.

Reply:
You're right though. It's not an Error if we are unable to obtain leadership.